rcl/test/rcl/test_guard_condition.cpp

```diff
@@ -202,6 +214,18 @@ TEST_F(
   EXPECT_EQ(RCL_RET_OK, ret);
   ret = rcl_guard_condition_fini(&guard_condition);
   EXPECT_EQ(RCL_RET_OK, ret);
+  // Try normal init and fini, but force an internal error on first try.
+  {
+    auto mock = mocking_utils::inject_on_return(
+      "lib:rcl", rmw_destroy_guard_condition, RMW_RET_ERROR);
+    ret = rcl_guard_condition_init(&guard_condition, &context, default_options);
```

Comment:
Shouldnt this init be left outside of the block that mocks the `rmw_destroy_guard_condition`?

Reply:
It can be outside, it doesn't have to be outside.